src/components/ngRedux.js

```diff
@@ -42,45 +48,56 @@ export default function ngReduxProvider() {
   };
 
   this.$get = ($injector) => {
-    const resolveMiddleware = middleware => isString(middleware)
-      ? $injector.get(middleware)
-      : middleware;
+    if (_providedStore) {
+      const emptyStore = createNgReduxStore($injector, [], [], state => state, undefined);
 
-    const resolvedMiddleware = map(_middlewares, resolveMiddleware);
+      return wrapStore(_providedStore, emptyStore);
+    }
 
-    const resolveStoreEnhancer = storeEnhancer => isString(storeEnhancer)
-      ? $injector.get(storeEnhancer)
-      : storeEnhancer;
+    return createNgReduxStore($injector, _middlewares, _storeEnhancers, _reducer, _initialState);
 
-    const resolvedStoreEnhancer = map(_storeEnhancers, resolveStoreEnhancer);
+  };
 
-    if(_reducerIsObject) {
-      const getReducerKey = key => isString(_reducer[key])
-        ? $injector.get(_reducer[key])
-        : _reducer[key];
+  this.$get.$inject = ['$injector'];
+}
 
-      const resolveReducerKey = (result, key) => assign({}, result,
-        { [key]: getReducerKey(key) }
-      );
+function createNgReduxStore($injector, _middlewares, _storeEnhancers, _reducer, _initialState) {
```

Comment:
By moving this function out, i'm getting `_reducerIsObject` is not defined, on line 77. Looks like it doesn't have the scope of all the vars defined at the top of `ngReduxProvider()`

Reply:
I resolved it for myself by moving `createNgReduxStore` inside of the `ngReduxProvider` method's scope but i'm not sure if that was a good solution or not.